File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/robotSpecific/targetLanguage/ev3dev/control_logic.py
#!/usr/bin/python

from __future__ import absolute_import
from roberta.ev3 import Hal
from ev3dev import ev3 as ev3dev
import math
import os
import time

class BreakOutOfALoop(Exception): pass
class ContinueLoop(Exception): pass

_brickConfiguration = {
    'wheel-diameter': 5.6,
    'track-width': 18.0,
    'actors': {
    },
    'sensors': {
    },
}
hal = Hal(_brickConfiguration)

___numberVar = 0
___booleanVar = True
___stringVar = ""
___colourVar = 'white'
___connectionVar = None
___numberList = [0, 0]
___booleanList = [True, True]
___stringList = ["", ""]
___colourList = ['white', 'white']
___connectionList = [___connectionVar, ___connectionVar]
def ____control():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    if ___booleanVar:
        pass
    elif ___booleanVar:
        pass
    if ___booleanVar:
        pass
    elif ___booleanVar:
        pass
    while True:
        pass
    for ___k0 in range(int(0), int(___numberVar), int(1)):
        pass
    for ___i in range(int(___numberVar), int(___numberVar), int(___numberVar)):
        pass
    while True:
        break
    while True:
        continue
    hal.waitFor(___numberVar)
    while ___booleanVar:
        pass
    while not ___booleanVar:
        pass
    for ___item in ___numberList:
        pass
    for ___item2 in ___booleanList:
        pass
    for ___item3 in ___stringList:
        pass
    for ___item4 in ___colourList:
        pass
    for ___item5 in ___connectionList:
        pass
    while True:
        if ___booleanVar:
            break
        if ___booleanVar:
            break
        hal.waitFor(15)
    while True:
        if ___booleanVar:
            break
        hal.waitFor(15)

def ____logic():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    hal.drawText(str(___numberVar == ___numberVar), ___numberVar, ___numberVar)
    hal.drawText(str(___numberVar != ___numberVar), ___numberVar, ___numberVar)
    hal.drawText(str(___numberVar < ___numberVar), ___numberVar, ___numberVar)
    hal.drawText(str(___numberVar <= ___numberVar), ___numberVar, ___numberVar)
    hal.drawText(str(___numberVar > ___numberVar), ___numberVar, ___numberVar)
    hal.drawText(str(___numberVar >= ___numberVar), ___numberVar, ___numberVar)
    hal.drawText(str(___booleanVar and ___booleanVar), ___numberVar, ___numberVar)
    hal.drawText(str(___booleanVar or ___booleanVar), ___numberVar, ___numberVar)
    hal.drawText(str(not ___booleanVar), ___numberVar, ___numberVar)
    hal.drawText(str(True), ___numberVar, ___numberVar)
    hal.drawText(str(False), ___numberVar, ___numberVar)
    hal.drawText(str(None), ___numberVar, ___numberVar)
    hal.drawText(str(___numberVar if ( ___booleanVar ) else ___numberVar), ___numberVar, ___numberVar)

def run():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    ____control()
    ____logic()

def main():
    try:
        run()
    except Exception as e:
        hal.drawText('Fehler im EV3', 0, 0)
        hal.drawText(e.__class__.__name__, 0, 1)
        hal.drawText(str(e), 0, 2)
        hal.drawText('Press any key', 0, 4)
        while not hal.isKeyPressed('any'): hal.waitFor(500)
        raise

if __name__ == "__main__":
    main()

File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/robotSpecific/targetLanguage/festobionic/control_logic.cpp
#define _ARDUINO_STL_NOT_NEEDED
#define LED_BUILTIN 13

#include <Arduino.h>

#include <NEPODefs.h>

void ____control();
void ____logic();

double ___numberVar;
bool ___booleanVar;
String ___stringVar;
std::list<double> ___numberList;
std::list<bool> ___booleanList;
std::list<String> ___stringList;
int _led_L = LED_BUILTIN;

void ____control() {
    if ( ___booleanVar ) {
    } else if ( ___booleanVar ) {
    }
    if ( ___booleanVar ) {
    } else if ( ___booleanVar ) {
    }
    while ( true ) {
        delay(1);
    }
    for (int ___k0 = 0; ___k0 < ___numberVar; ___k0 += 1) {
        delay(1);
    }
    for (int ___i = ___numberVar; ___i < ___numberVar; ___i += ___numberVar) {
        delay(1);
    }
    while ( true ) {
        break;
        delay(1);
    }
    while ( true ) {
        continue;
        delay(1);
    }
    delay(___numberVar);
    while ( ! ___booleanVar ) {
        delay(1);
    }
    while ( ___booleanVar ) {
        delay(1);
    }
    for ( double ___item : ___numberList ) {
        delay(1);
    }
    for ( bool ___item2 : ___booleanList ) {
        delay(1);
    }
    for ( String ___item3 : ___stringList ) {
        delay(1);
    }
    while (true) {
        if ( ___booleanVar ) {
            break;
        }
        if ( ___booleanVar ) {
            break;
        }
        delay(1);
    }
    while (true) {
        if ( ___booleanVar ) {
            break;
        }
        delay(1);
    }
}

void ____logic() {
    Serial.println(___numberVar == ___numberVar);
    Serial.println(___numberVar != ___numberVar);
    Serial.println(___numberVar < ___numberVar);
    Serial.println(___numberVar <= ___numberVar);
    Serial.println(___numberVar > ___numberVar);
    Serial.println(___numberVar >= ___numberVar);
    Serial.println(___booleanVar && ___booleanVar);
    Serial.println(___booleanVar || ___booleanVar);
    Serial.println(! ___booleanVar);
    Serial.println(true);
    Serial.println(false);
    Serial.println(NULL);
    Serial.println(( ( ___booleanVar ) ? ( ___numberVar ) : ( ___numberVar) ));
}

void setup()
{
    Serial.begin(9600);
    pinMode(_led_L, OUTPUT);
    ___numberVar = 0;
    ___booleanVar = true;
    ___stringVar = "";
    ___numberList = {0, 0};
    ___booleanList = {true, true};
    ___stringList = {"", ""};
}

void loop()
{
    ____control();
    ____logic();
}